crypto/src/memo.rs

```diff
@@ -55,7 +62,7 @@ pub struct MemoCiphertext(pub [u8; MEMO_CIPHERTEXT_LEN_BYTES]);
 impl MemoPlaintext {
     /// Encrypt a memo, returning its ciphertext.
     pub fn encrypt(&self, memo_key: PayloadKey) -> MemoCiphertext {
```

Comment:
If we eliminate the `MemoPlaintext` entirely, we won't have a struct to implement this on, so it'll need to be a method on `MemoCiphertext`, and we'll need it to be fallible, because we want to error when the string is too long:
```rust
impl MemoCiphertext {
    pub fn encrypt(memo: &str, memo_key: PayloadKey) -> Result<Self, anyhow::Error> {
        // ..
    }
}
```